sources/imdsclient/src/lib.rs

```diff
@@ -21,20 +21,27 @@ use log::{debug, info, trace, warn};
 use reqwest::Client;
 use serde_json::Value;
 use snafu::{ensure, OptionExt, ResultExt};
-use std::time::Duration;
-use tokio::time;
+use tokio::time::{sleep, timeout, Duration};
+use tokio_retry::{strategy::FibonacciBackoff, Retry};
 
 const BASE_URI: &str = "http://169.254.169.254";
 const PINNED_SCHEMA: &str = "2021-01-03";
 
 // Currently only able to get fetch session tokens from `latest`
 const SESSION_TARGET: &str = "latest/api/token";
 
+fn retry_token_strategy() -> impl Iterator<Item = Duration> {
+    FibonacciBackoff::from_millis(1000)
```

Comment:
```suggestion
    FibonacciBackoff::from_millis(500)
```
1s seems kind of high for a local service that's in the boot path.